src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -356,6 +356,51 @@ class spell_sha_crash_lightning : public SpellScript
     size_t _targetsHit = 0;
 };
 
+// 378270 - Deeply Rooted Elements
+class spell_sha_deeply_rooted_elements : public AuraScript
+{
+    PrepareAuraScript(spell_sha_deeply_rooted_elements);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_LAVA_BURST, SPELL_SHAMAN_ASCENDANCE_ELEMENTAL, SPELL_SHAMAN_ASCENDANCE_ENHANCEMENT, SPELL_SHAMAN_ASCENDANCE_RESTORATION })
+            && ValidateSpellEffect({ { spellInfo->Id, EFFECT_0 } });
+    }
+
+    bool CheckProc(AuraEffect const* /*aurEff*/, ProcEventInfo& procInfo)
+    {
+        if (procInfo.GetSpellInfo()->Id == SPELL_SHAMAN_LAVA_BURST && procInfo.GetActor()->ToPlayer()->GetPrimarySpecialization() != TALENT_SPEC_SHAMAN_ELEMENTAL)
+            return false;
+
+        return true;
+    }
+
+    void HandleProc(AuraEffect* aurEff, ProcEventInfo& eventInfo)
+    {
+        Unit* target = eventInfo.GetActor();
+
+        uint32 ascendanceSpecSpell = 0;
+        switch (target->ToPlayer()->GetPrimarySpecialization())
+        {
+            case TALENT_SPEC_SHAMAN_ELEMENTAL:   ascendanceSpecSpell = SPELL_SHAMAN_ASCENDANCE_ELEMENTAL; break;
+            case TALENT_SPEC_SHAMAN_ENHANCEMENT: ascendanceSpecSpell = SPELL_SHAMAN_ASCENDANCE_ENHANCEMENT; break;
+            case TALENT_SPEC_SHAMAN_RESTORATION: ascendanceSpecSpell = SPELL_SHAMAN_ASCENDANCE_RESTORATION; break;
+            default:
+                break;
+        }
+        target->CastSpell(target, ascendanceSpecSpell, true);
+
+        if (Aura* ascendanceAura = target->GetAura(ascendanceSpecSpell))
```

Comment:
you just cast the spell before, this would increase it by the dummy amount anyway, just use the impl i gave